services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/MetersValidationSingleSwFlowSpec.groovy

```diff
@@ -0,0 +1,411 @@
+        def defaultMeters = northbound.getAllMeters(sw.dpId)
+
+        when: "Create a flow"
+        def flow = flowHelper.addFlow(flowHelper.singleSwitchFlow(sw))
+        def metersIds = getCreatedMetersId(sw.dpId, defaultMeters)
+        def createdCookies = getCreatedCookies(sw.dpId, metersIds)
+
+        and: "Change cookies for the created flow directly in DB"
+        def newFlowCookies = [createdCookies[0] + 1, createdCookies[1] + 1]
+        /** at this point rules are set for given flow. Now update flow cookie directly via DB,
+         it is done just for moving rules from the proper section into the excess section*/
+        database.updateFlowCookie(flow.id, newFlowCookies[0], newFlowCookies[1])
+
+        then: "Rule validate info(original cookies) is moved into the excess section"
+        def switchValidateInfo = northbound.switchValidate(sw.dpId)
+        switchValidateInfo.rules.excess.collect { it }.containsAll(createdCookies)
+
+        and: "Rule validate info(new cookies) is moved into the excess section"
+        switchValidateInfo.rules.missing.collect { it }.containsAll(newFlowCookies)
+        ruleSectionsAreEmpty(switchValidateInfo, ["proper"])
+
+        when: "Delete the flow"
+        flowHelper.deleteFlow(flow.id)
+
+        then: "Check that rules validate info is also deleted"
+        Wrappers.wait(WAIT_OFFSET) {
+            def switchValidateInfoAfterDelete = northbound.switchValidate(sw.dpId)
+            ruleSectionsAreEmpty(switchValidateInfoAfterDelete)
+            meterSectionsAreEmpty(switchValidateInfoAfterDelete)
+        }
+
+        where:
+        switchType   | switches
+        "Centec"     | getCentecSwitches()
+        "non-Centec" | getNonCentecSwitches()
+    }
+
+    def "Not able to get the switch validate info on a NOT supported switch"() {
+        given: "Not supported switch"
+        def switches = topology.activeSwitches.findAll { it.ofVersion == "OF_12" }
+        assumeTrue("Unable to find required switches in topology", switches as boolean)
+        def sw = switches.first()
+
+        when: "Try to invoke the switch validate request"
+        northbound.switchValidate(sw.dpId)
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+        exc.responseBodyAsString.to(MessageError).errorMessage ==
+                "Meters are not supported on switch $sw.dpId because of OF version OF_12"
+    }
+
+    @Memoized
+    List<Switch> getNonCentecSwitches() {
+        topology.activeSwitches.findAll { !it.centec && it.ofVersion == "OF_13" }
+    }
+
+    @Memoized
+    List<Switch> getCentecSwitches() {
+        topology.getActiveSwitches().findAll { it.centec }
+    }
+
+    @Memoized
+    List<Switch> getSwitchNotSupportedMeter() {
+        topology.activeSwitches.findAll { it.ofVersion == "OF_12" }
+    }
+
+    List<Integer> getCreatedMetersId(switchId, defaultMeters) {
```

Comment:
I think the proper way to retrieve non-default meters is to perform
```
northbound.getAllMeters(switchId).meterEntries.findAll { it.meterId > MAX_SYSTEM_RULE_METER_ID }
```

Reply:
method was rewrited

Reply:
Fixed